src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -127,6 +130,24 @@ void leader_balancer::on_leadership_change(
     }
 }
 
+void leader_balancer::on_maintenance_change(
+  model::node_id, model::maintenance_state ms) {
+    if (!_enabled()) {
+        return;
+    }
+
+    if (!_raft0->is_elected_leader()) {
+        return;
+    }
+
+    // if a node transitions out of maintenance wake up the balancer early to
+    // transfer leadership back to it.
+    if (ms == model::maintenance_state::inactive) {
+        _timer.cancel();
+        _timer.arm(leader_activation_delay);
```

Comment:
presumably we don't need to unmute because the leader balancer recomputes muted nodes on demand?

Reply:
Yep!